pmd-java/src/main/resources/category/java/codestyle.xml

```diff
@@ -1353,6 +1353,54 @@ public class ClassInDefaultPackage {
         </example>
     </rule>
 
+    <rule name="UseUnderscoresInNumericLiterals"
+          language="java"
+          since="6.10.0"
+          minimumLanguageVersion="1.7"
+          message="Number {0} should separate every third digit with an underscore"
+          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_codestyle.html#useunderscoresinnumericliterals">
+        <description>
+            Numeric literals with more than 3 digits must use '_' as a separator.
+        </description>
+        <priority>3</priority>
+        <properties>
+            <property name="version" value="2.0"/>
+            <property name="acceptableDecimalLength" type="Integer" value="4" min="3" max="1000"
+                      description="Maximum acceptable length within which the rule is not applicable"/>
+            <property name="xpath">
+                <value>
+                    <![CDATA[
+//Literal[@IntLiteral = true() or
+ @LongLiteral = true() or
+ @DoubleLiteral = true() or
+ @FloatLiteral = true()]
+ [ not (matches(@Image, "^0([xb]?[0-9a-fA-F]+)?$"))]
```

Comment:
I think this may be simplified as `^0[xb]?.*`

Reply:
I thought of this initially but then it does not hold good with values like 0.12345.
It does not catch issues with values - 0.[0-9]+

Reply:
Oh ok! I think maybe we may be as bold as using `^0[^.]` then?

Reply:
yes, this does the trick!